internal/clusterfeature/featuremanager.go

```diff
@@ -0,0 +1,150 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusterfeature
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/goph/emperror"
+	"github.com/goph/logur"
+	"github.com/goph/logur/adapters/logrusadapter"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// syncFeatureManager synchronous feature manager
+type syncFeatureManager struct {
+	logger         logur.Logger
+	clusterService ClusterService
+	helmService    HelmService
+}
+
+// NewSyncFeatureManager builds a new feature manager component
+func NewSyncFeatureManager(clusterService ClusterService) FeatureManager {
+	l := logur.WithFields(logrusadapter.New(logrus.New()), map[string]interface{}{"component": "feature-manager"})
+	return &syncFeatureManager{
+		logger:         l,
+		clusterService: clusterService,
+		helmService: &featureHelmService{ // wired private component!
+			logger: logur.WithFields(l, map[string]interface{}{"comp": "helm-installer"}),
+		},
+	}
+}
+
+func (sfm *syncFeatureManager) Activate(ctx context.Context, clusterId uint, feature Feature) error {
+
+	cluster, err := sfm.clusterService.GetCluster(ctx, clusterId)
+	if err != nil {
+		// internal error at this point
+		return emperror.WrapWith(err, "failed to activate feature")
+	}
+
+	ns, ok := feature.Spec[DNSExternalDnsNamespace]
```

Comment:
`syncFeatureManager` seems like a generic thing, so why does it have ExternalDNS specific stuff hardcoded?